src/test/java/com/teragrep/rlp_03/client/ClientTest.java

```diff
@@ -0,0 +1,222 @@
+        // TODO perhaps <Integer, Future<Something>> ?
+        // FIXME what should be Something, RelpFrame is immediately deallocated after FrameDelegate
+        // TODO design better: Futures are not optimal for multi-complete/disruptor pattern
+        HashMap<Integer, CompletableFuture<String>> pendingTransactions = new HashMap<>();
+
+        FrameDelegate essentialClientDelegate = new FrameDelegate() {
+
+            @Override
+            public boolean accept(FrameContext frameContext) {
+                LOGGER.debug("client got <[{}]>", frameContext.relpFrame());
+
+                int txn = frameContext.relpFrame().txn().toInt();
+
+                // TODO implement better handling for hint frames
+                if (txn == 0) {
+                    return true;
+                }
+
+                CompletableFuture<String> future = pendingTransactions.remove(txn);
+
+                if (future == null) {
+                    throw new IllegalStateException("txn not pending <[" + txn + "]>");
+                }
+
+                future.complete(frameContext.relpFrame().payload().toString());
+                LOGGER.debug("completed transaction for <[{}]>", txn);
+
+                frameContext.relpFrame().close();
+
+                return true;
+            }
+
+            @Override
+            public void close() throws Exception {
+                LOGGER.debug("client FrameDelegate close");
+            }
+
+            @Override
+            public boolean isStub() {
+                return false;
+            }
+        };
+
+        ConnectContext connectContext = connectContextFactory
+                .create(new InetSocketAddress(port), essentialClientDelegate, connectionContextConsumer);
+        connectContext.register(server.eventLoop);
+
+        try (ConnectionContext connectionContext = readyContextFuture.get()) {
+            Transmit transmit = new Transmit(connectionContext, pendingTransactions);
+
+            CompletableFuture<String> open = transmit
+                    .transmit("open", "a hallo yo client".getBytes(StandardCharsets.UTF_8));
+            CompletableFuture<String> syslog = transmit
+                    .transmit("syslog", "yonnes payload".getBytes(StandardCharsets.UTF_8));
+            CompletableFuture<String> close = transmit.transmit("close", "".getBytes(StandardCharsets.UTF_8));
+
+            try {
+                String openResponse = open.get();
+                LOGGER.debug("openResponse <[{}]>", openResponse);
+                String syslogResponse = syslog.get();
+                LOGGER.debug("syslogResponse <[{}]>", syslogResponse);
+                Assertions.assertEquals("200 OK", syslogResponse);
+                String closeResponse = close.get();
+                LOGGER.debug("closeResponse <[{}]>", closeResponse);
+            }
+            catch (ExecutionException executionException) {
+                throw new RuntimeException(executionException); // TODO
+            }
```

Comment:
Don't try/catch, use assertions instead

Reply:
unwrapped, last is required because ExecutionException is a checked type